executor/executor.go

```diff
@@ -278,8 +281,15 @@ func (e *Executor) handleLaunch(ev *executor.Event) error {
 	// Update status to RUNNING
 	status := e.newStatus(task.GetTaskID())
 	status.State = mesos.TASK_RUNNING.Enum()
+	err = e.updateStatus(status)
+	if err != nil {
+		return e.throwError(task.GetTaskID(), err)
+	}
 
-	return e.updateStatus(status)
+	// Handle the container exit
+	go e.waitContainer(e.ContainerTasks[task.TaskID], task.GetTaskID())
```

Comment:
This  `waitContainer` handle he nominal case and the one line 261 handle the case when the container exit quickly before we could execute post-run hooks. 

So in nominal case we have 2 goroutines waiting for the container to exit to do the tearDown. Do we need some kind of synchronization here to avoid doing the tearDown twice ?

Reply:
I removed the one line 261, I just forgot to remove it. The waitContainer function call should be done AFTER the TASK_RUNNING update status in order to avoid to send a TASK_FINISHED (from waitContainer goroutine) before the TASK_RUNNIN update.